pytrip/vdx.py

```diff
@@ -867,19 +867,15 @@ def read_dicom(self, info, data):
         else:
             contours = data.ContourSequence
         for i, contour in enumerate(contours):
-            key = int((float(contour.ContourData[2]) - min(self.cube.slice_pos)) * 100)
-            if key not in self.slices:
-                self.slices[key] = Slice(cube=self.cube)
-                self.slice_z.append(key)
-            self.slices[key].add_dicom_contour(contour)
 
-    def get_thickness(self):
-        """
-        :returns: thickness of slice in [mm]. If there is only one slice, 3 mm is returned.
-        """
-        if len(self.slice_z) <= 1:
-            return 3  # TODO: what is this? And shoudn't it be float?
-        return abs(float(self.slice_z[1]) - float(self.slice_z[0])) / 100
+            # get current slice position
+            _z_pos = contour.ContourData[2]
+
+            # if we have a new z_position, add a new slice object to self.slices
+            if _z_pos not in [i.get_position() for i in self.slices]:
+                logger.debug("Append new slice at z_position {:f} to slices list:".format(_z_pos))
+                self.slices.append(Slice(cube=self.cube))
+                self.slices[-1].add_dicom_contour(contour)
```

Comment:
This looks a bit enigmatic, what about following code ?
```
s = Slice(cube=self.cube)
s.add_dicom_contour(contour)
self.slices.append(s)
```

I think it is a bit easier to debug.